jaeger/cmd/check.go

```diff
@@ -0,0 +1,226 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+
+	// jaegerExtensionName is the name of jaeger extension
+	jaegerExtensionName = "linkerd-jaeger"
+
+	// linkerdJaegerExtensionCheck adds checks related to the jaeger extension
+	linkerdJaegerExtensionCheck healthcheck.CategoryID = jaegerExtensionName
+)
+
+var (
+	jaegerNamespace string
+)
+
+type checkOptions struct {
+	wait   time.Duration
+	output string
+}
+
+func jaegerCategory() (*healthcheck.Category, error) {
+
+	kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	checkers := []healthcheck.Checker{}
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector and jaeger service account exists", "l5d-jaeger-sc-exists", true, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Collector Service Account
+				return healthcheck.CheckServiceAccounts(ctx, kubeAPI, []string{"collector", "jaeger"}, jaegerNamespace, "")
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("collector config map exists", "l5d-jaeger-oc-cm-exists", false, true, time.Time{}, false).
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger Service Account
+				_, err = kubeAPI.CoreV1().ConfigMaps(jaegerNamespace).Get(ctx, "collector-config", metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}))
+
+	checkers = append(checkers,
```

Comment:
Should we add a retryDeadline to these checks which look for certain pods to be running so that they retry and we get the spinner behavior that the `linkerd check` command does?

Reply:
Good Find! :)